src/test/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProviderTest.java

```diff
@@ -0,0 +1,279 @@
+    private static final String[] DELETE_COMMAND_PATTERN = new String[]{"scontrol DELETE "
+            + "PartitionName=deletedPartiiton"};
+    private static final String[] UPDATE_COMMAND_PATTERN = new String[]{"scontrol UPDATE "
+            + "PartitionName=updatingPartitionName AllowGroups=ALL Nodes=worker1,worker2"};
+    private static final String[] SINFO_COMMAND_PATTERN = new String[]{"sinfo --partition=updatingPartitionName "
+            + " -h -o \"%R|%N|%c|%g\""};
+    private static final String SCONTROL_COMMAND_TEMPLATE = "scontrolCommand";
+    private static final String SINFO_COMMAND_TEMPLATE = "sinfo";
+    private static final List<String> ownerList = List.of("root", "slurm");
+    private static final List<String> parallelEnvList = List.of("someVar1", "someVar1");
+    private static final List<String> nodeListOneNode = List.of("worker1");
+    private static final List<String> nodeListTwoNodes = new ArrayList<>(List.of("worker1", "worker2"));
+    private static final List<String> groupsList = new ArrayList<>(List.of("ALL"));
+
+    private final Map<String, Integer> slotDescriptionOneNode = Map.of("worker1", 1);
+
+    @Autowired
+    private SlurmQueueProvider slurmQueueProvider;
+
+    @MockBean
+    private SimpleCmdExecutor mockCmdExecutor;
+
+    @MockBean
+    private GridEngineCommandCompiler mockCommandCompiler;
+
+    @ParameterizedTest
+    @MethodSource("incorrectDataForPartitionCreateAndUpdate")
+    public void registerAndUpdatePartitionIncorrectDataException(final QueueVO incorrectRequest) {
+        final GridEngineException gridExceptionCreate = Assertions.assertThrows(GridEngineException.class,
+                () -> slurmQueueProvider.registerQueue(incorrectRequest));
+        Assertions.assertNotNull(gridExceptionCreate.getMessage());
+
+        final GridEngineException gridExceptionUpdate = Assertions.assertThrows(GridEngineException.class,
+                () -> slurmQueueProvider.updateQueue(incorrectRequest));
+        Assertions.assertNotNull(gridExceptionUpdate.getMessage());
+    }
+
+    @Test
+    public void registerPartitionSuccessful() {
+        final CommandResult creationResult = CommandResult.builder()
+                .stdOut(validStdout)
+                .stdErr(Collections.emptyList())
+                .exitCode(0).build();
+        doReturn(CREATE_COMMAND_PATTERN).when(mockCommandCompiler).compileCommand(Mockito.eq(CommandType.SLURM),
+                Mockito.matches(SCONTROL_COMMAND_TEMPLATE), Mockito.any());
+        doReturn(creationResult).when(mockCmdExecutor).execute(CREATE_COMMAND_PATTERN);
+
+        final QueueVO correctRequest = QueueVO.builder()
+                .name("newPartition")
+                .hostList(nodeListOneNode)
+                .allowedUserGroups(groupsList)
+                .build();
+        final Queue createdPartitionEntity = Queue.builder()
+                .name("newPartition")
```

Comment:
we could unify the duplicate values into a separate static variable
in several places as well

Reply:
done